Enchantments+-/src/main/java/org/eps/pvppack/EnchantProcessor.java

```diff
@@ -229,7 +260,31 @@ public static ItemStack getHead(LivingEntity e)
 			else
 				return new ItemStack(Material.matchMaterial("SKULL_HEAD"), 1, (short) 5);
 		default:
+			if (mobHeads.containsKey(e.getType())) {
+				return getCustomSkull(mobHeads.get(e.getType()), e.getName() + " Head");
+			}
 			return null;
 		}
 	}
+
+	private static ItemStack getCustomSkull(String texture, String name) {
+		ItemStack head = LegacyUtil.isLegacy() ? new ItemStack(Material.matchMaterial("SKULL_ITEM"), 1) : new ItemStack(Material.PLAYER_HEAD, 1);
+
+		SkullMeta skull = (SkullMeta) head.getItemMeta();
+		skull.setDisplayName(name);
+		GameProfile profile = new GameProfile(UUID.randomUUID(), null);
+
+		profile.getProperties().put("textures", new Property("textures", texture));
+
+		try {
+			Method mtd = skull.getClass().getDeclaredMethod("setProfile", GameProfile.class);
```

Comment:
Maybe it would be better to cache the returned object (of the line I marked or the method as a whole - latter might be better if possible thanks to EnumMaps) as reflection is relatively expensive as far as I know, but it always depends on how often it is used - so caching would bring the effects to a minimum.